### ISSUE TITLE
Write/Improve documentation on block creation in Woo Blocks codebase

### ISSUE THREAD

**Issue by nerrad** [woocommerce/woocommerce-gutenberg-products-block#4626]
There's a number of missing documentation in our codebase:

* [ ] creating a new Woo Block
* [ ]  create a new Product Element Block
* [ ]  create a new innerBlocks Block (and register).

While there is WordPress documentation geared towards more generic "create a block", that's geared more towards creating blocks from scratch. Within Woo Blocks, we've got some additional abstractions and affordances to account for our build process and various extensibility requirements which WordPress generic docs could never cover completely. Thus, there's some importance in ensuring this is documented not only for internal team needs as new team members join us (or we from 3rd party contributors), but also for those extending/building blocks within the same ecosystem (especially in the case of registering inner Blocks.

This issue is to capture this need and ensure we've accounted for this in planning.

**Comment by mikejolley** [comment]
These docs cover block/inner block registration: https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/packages/checkout/blocks-registry

**Comment by mikejolley** [comment]
@nerrad This looks like something more suited to cooldown and could use a better outline on things to cover, what level is assumed, and how deep to go. We do have some docs scattered, and things like "Product Elements" ideally need updating to match the cart/checkout inner block structure before we commit to formal documentation. Thoughts?

**Comment by nerrad** [comment]
WRT to product elements, I do agree that there needs to be some decisions/work there before doing any documentation around those blocks specifically. However, I wonder if it would make sense to continue the work Thomas has done on the scaffolding tool for creating either Woo Blocks or extensions that hook into available extensibility interfaces and create some tutorials around that (eg. tutorial for creating a payment method integration etc).

So essentially repurpose this issue to:

- finish and launch the cli scaffolding tool for existing extensibility integrations.
- write some tutorials: potential examples could be - "How to build a payment method integration?", "How to create a new inner block for Cart and Checkout Blocks/All Product blocks?".

In other words, what are some things we can create more of right now to make it easier for extension developers to utilize the interfaces we have available now?

I'm okay with this being punted to the cooldown but it also is needed work for making Cart and Checkout blocks more accessible for extension developers integrating their extensions.

**Comment by ralucaStan** [comment]
I am going to close this ticket as we have made some progress in some of the directions mentioned here and I feel we would need to have specific tickets for each initiative. Another reason is that @manospsyx and his team also have planned to write more guides, which make this ticket obsolete.

> finish and launch the cli scaffolding tool for existing extensibility integrations.
- [Add @woocommerce/extend-cart-checkout-block package #32323](https://github.com/woocommerce/woocommerce/pull/32323) was merged in WC core and we are waiting for it to be released on npm. It might be worth adding an inner blocks to this tool as a cooldown project. CC @opr 

> write some tutorials: potential examples could be - "How to build a payment method integration?", "How to create a new inner block for Cart and Checkout Blocks/All Product blocks?".


We have a wonderful guide [Hiding Shipping and Payment Options in the Cart and Checkout Blocks](https://developer.woocommerce.com/2022/05/20/hiding-shipping-and-payment-options-in-the-cart-and-checkout-blocks/)  by @manospsyx 

We also have [this repo ](https://github.com/woocommerce/newsletter-test) which is an example of integrating with inner blocks in the WooCommerce Blocks Checkout
